### DISCUSSION TITLE
FormState

### DISCUSSION THREAD

**Discussion by ghost** [react-hook-form/react-hook-form#10875] [General] [answered]
I have a form that should be sent to the server in two stages, the first step is that the user has to fill in the forms and click the test button, when the test button is clicked if the test was successful, the test button gets disabled and the save button appears, now I want the the test button to show up again if the user modifies any of the fields prior clicking the save button and the save button gets disabled again. Have can I achieve this behavior if the form was dirty already ? how can I know that the user has modified the form after testing when the form is already dirty because it's filled with data ?

**Comment by Moshyfawn** [comment, marked as answer]
You can call [the `reset` method](https://react-hook-form.com/docs/useform/reset) on your "test" button click with your current form values and `keepValues: true`. This way you set new default form values if the test's been successful.

```jsx
function Form() {
  const { register, reset, getValues, formState } = useForm({
    defaultValues: {
      name: "John Doe",
      age: "30"
    }
  })

  const { isDirty } = formState

  const handleTest = () => {
    if (testForm()) {
      reset(getValues(), { keepValues: true })
      return
    }
  }

  return (
    <form>
      <input {...register("name")} />
      <input {...register("age")} />
      {isDirty && (
        <button type="button" onClick={handleTest}>
          Save
        </button>
      )}
      <button type="submit">Submit</button>
    </form>
  )
}
```

**Reply by ghost** [reply]
Thanks!
